### ISSUE TITLE
Deprecated rules shouldn't run if their WCAG tag is configured

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4523]
The `aria-roledescription` rule was deprecated a while ago. In axe-core this was disabled and tagged as deprecated. That should have turned it off in all tools that use axe-core. It has done so in axe extension, but because of the way tags work in axe-core, it didn't turn the rule of the various axe SDKs (as well as anything built on top of them).

We'll want the following to be true:

```js
axe.run({ runOnly: ['wcag2a'] }) // does **not** run aria-roledescription
axe.run({ runOnly: ['wcag2a', 'deprecated' }) // Runs aria-roledescription
axe.run({ rules: { rules: { "aria-roledescription": { enabled: true } } } }) // Runs aria-roledescription
```

This should make deprecated rules work the same as experimental rules. And since it'll work the same way, we can use the same mechanism. We should be able to do this by adding "deprecated" to the [tagExclude option](https://github.com/dequelabs/axe-core/blob/develop/lib/core/base/audit.js#L194).
